SecretSanta/test/SecretSanta.Business.Tests/Controllers/UserRepositoryTests.cs

```
@@ -0,0 +1,105 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System.Linq;
+using SecretSanta.Api.Controllers;
+using System.Collections.Generic;
+using SecretSanta.Business;
+using SecretSanta.Data;
+using Microsoft.AspNetCore.Mvc;
+using System;
+
+namespace SecretSanta.Business.Tests.Controllers
+{
+    [TestClass]
+    public class UserRepositoryTests{
+
+        [TestMethod]
+        public void Create_WithGoodData_ReturnsListWithUser(){
+            UserRepository repo = new UserRepository();
+            
+            User newUser = new User();
+            newUser.FirstName = "New";
+            newUser.LastName = "Mann";
+            
+            User created = repo.Create(newUser);
+            Assert.AreEqual(newUser, created);
+        }
+        [TestMethod]
+         public void Create_WithGoodData_ReturnsListWithoutUser(){
+            UserRepository repo = new UserRepository();
+            
+            User newUser = new User();
+            newUser.FirstName = "New";
+            newUser.LastName = "Mann";
+            
+            User created = repo.Create(newUser);
+            int id = created.Id;
+            Assert.IsTrue(!(repo.GetItem(id)==newUser));
+        }
+        [TestMethod]
+        public void GetItem_BadID_ThrowsExcept(){
+            UserRepository repo = new UserRepository();
+            Boolean res = false;
+            try{
+                repo.GetItem(-1);
+            }catch(ArgumentOutOfRangeException e){
+                res = true;
+            }
+            Assert.IsTrue(res);
+        }
+
+        [TestMethod]
+        [DataRow(3)]
+        public void GetItem_GoodID_ReturnsUser(int id){
+            UserRepository repo = new UserRepository();
+            User test = null;
+            Boolean res = true;
+            try{
+                test = repo.GetItem(id);
+            }catch(ArgumentOutOfRangeException e){
```

Given that this test is expected to be testing a good code path I would not expect to need a `try/catch` block here. In general simply let the test throw the exception if it occurs. A failing test is any test that throw an exception so if that cases happens your test will appropriately fail.